src/main/java/io/mosip/mimoto/controller/V2IssuersController.java

```diff
@@ -0,0 +1,163 @@
+            logger.error("Exception occurred while fetching issuers ", e);
+            responseWrapper.setErrors(List.of(new ErrorDTO(API_NOT_ACCESSIBLE_EXCEPTION.getCode(), API_NOT_ACCESSIBLE_EXCEPTION.getMessage())));
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseWrapper);
+        }
+
+        return ResponseEntity.status(HttpStatus.OK).body(responseWrapper);
+    }
+
+    @GetMapping("/{issuer-id}/credentials-supported")
+    public ResponseEntity<Object> getCredentialsSupportedForIssuer(@PathVariable("issuer-id") String issuerId,
+                                                                   @RequestParam(required = false) String search) {
+        ResponseWrapper<Object> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setId(ID);
+        responseWrapper.setVersion("v2");
+        responseWrapper.setResponsetime(DateUtils.getRequestTimeString());
+        IssuerSupportedCredentialsResponse credentialsSupported;
+        try {
+            credentialsSupported = v2IssuersService.getCredentialsSupported(issuerId, search);
+        }catch (ApiNotAccessibleException | IOException exception){
+            logger.error("Exception occurred while fetching issuers ", exception);
+            responseWrapper.setErrors(List.of(new ErrorDTO(API_NOT_ACCESSIBLE_EXCEPTION.getCode(), API_NOT_ACCESSIBLE_EXCEPTION.getMessage())));
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseWrapper);
+        }
+        responseWrapper.setResponse(credentialsSupported);
+
+        if (credentialsSupported.getSupportedCredentials().isEmpty()) {
+            logger.error("invalid issuer id passed - {}", issuerId);
+            responseWrapper.setErrors(List.of(new ErrorDTO(INVALID_ISSUER_ID_EXCEPTION.getCode(), INVALID_ISSUER_ID_EXCEPTION.getMessage())));
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(responseWrapper);
+        }
+
+        return ResponseEntity.status(HttpStatus.OK).body(responseWrapper);
+    }
+
+    @PostMapping(value = "/get-token/issuer", consumes = {MediaType.APPLICATION_FORM_URLENCODED_VALUE})
+    public ResponseEntity getToken(@RequestParam Map<String, String> params, @RequestParam String issuer) {
+        logger.debug("Started Token Call get-token-> " + params.toString());
+        try {
+            IssuerDTO issuerDTO = null;
+            Reader reader = new InputStreamReader(Objects.requireNonNull(this.getClass()
+                    .getResourceAsStream("/v2-issuers-config.json")));
+            IssuersDTO issuersDto = new Gson().fromJson(reader, IssuersDTO.class);
+            Optional<IssuerDTO> issuerConfigResp = issuersDto.getIssuers().stream()
+                    .filter(issuerResponse -> issuerResponse.getCredential_issuer().equals(issuer))
+                    .findFirst();
+            if (issuerConfigResp.isPresent()) issuerDTO = issuerConfigResp.get();
```

Comment:
Instead of reading config file we should utilise the utility written for getting issuer config

Reply:
fixed